### PR TITLE
Replicate compiler errors in decoders on JDK17 for "yield()" #910

### PR DESCRIPTION
**Pull request by writeoncereadmany**
This currently successfully builds in JDK 8 and 11, but on JDK 17, we get the following output:

```
tom.johnson@tom-johnson-laptop-2:~/dev/simple-binary-encoding$ BUILD_JAVA_VERSION=17 ./gradlew test
/home/tom.johnson/dev/simple-binary-encoding/sbe-tool/build/generated-src/issue910/Issue910Decoder.java:229: error: invalid use of a restricted identifier 'yield'
        builder.append(yield());
                       ^
  (to invoke a method called yield, qualify the yield with a receiver or type name)
1 error

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':sbe-tool:compileGeneratedJava'.
> Compilation failed; see the compiler error output for details.
```

### CODE REVIEW

**Comment by mjpt777** [comment]
The approach looks fine. Can you please add a test that loads the schema directly rather than have have it in the gradle build file?

**Comment by writeoncereadmany** [comment]
Switched to a unit test that loads and compiles the schema directly, that's a much better approach thanks!

Note: if you undo my fix, this still only fails when run with a Java 17 toolchain - I couldn't see an easy way to insist on that in the test. But adding a JDK17 dependency for the test for those running pre-JDK17 seems wrong, so I think this is probably the best approach.

**Comment by mjpt777** [comment]
A test can be specified to run with a specific version. https://github.com/real-logic/aeron/blob/master/aeron-system-tests/src/test/java/io/aeron/cluster/FailedFirstElectionClusterTest.java#L78

**Comment by mjpt777** [comment]
You will need to generate the new IR codecs as part of this change.

**Comment by writeoncereadmany** [comment]
Codecs regenerated and test is set to only run against JDK17 and higher.

**Comment by writeoncereadmany** [comment]
everything's green now, thanks for your pointers! :)

File: sbe-tool/src/test/java/uk/co/real_logic/sbe/generation/java/QualifiedYieldTest.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    private static final Class<DirectBuffer> READ_ONLY_BUFFER_CLASS = DirectBuffer.class;
+    private static final String READ_ONLY_BUFFER_NAME = READ_ONLY_BUFFER_CLASS.getName();
+
+    private final StringWriterOutputManager outputManager = new StringWriterOutputManager();
+
+    @Test
+    @EnabledForJreRange(min = JRE.JAVA_17)
+    void shouldGenerateValidJava() throws Exception
+    {
+        System.setProperty(SbeTool.KEYWORD_APPEND_TOKEN, "_");
```
[the remark this answers is not in the document]
Comment (writeoncereadmany) [reply sbe-tool/src/test/java/uk/co/real_logic/sbe/generation/java/QualifiedYieldTest.java]: Nope! That's even actually potentially problematic, if yield were ever to be described as a keyword. Removed.

**Comment by mjpt777** [comment]
The build is failing. Please remove unused imports.

**Comment by writeoncereadmany** [comment]
My bad, I'm keeping an eye on the build but I haven't gotten used to the workflow for this project yet. And hopefully I won't have to :)

**Pull request merged**